File: pipeline/kite9-pipeline-common/src/main/java/com/kite9/k9server/command/xml/replace/ReplaceTagUrl.java
package com.kite9.k9server.command.xml.replace;

import org.w3c.dom.Element;

import com.kite9.k9server.adl.holder.pipeline.ADLDom;

/**
 * Allows the to field to be loaded from a url.
 * 
 * @author robmoffat
 *
 */
public class ReplaceTagUrl extends ReplaceTag {

	@Override
	protected Element getToContent(ADLDom adl) {
		Element out = getForeignElementCopy(adl.getDocument(), adl.getUri(), to, false, adl);
		return out;
	}

	
}

File: pipeline/kite9-pipeline-common/src/main/kotlin/com/kite9/pipeline/command/xml/replace/ReplaceTagUrl.kt
package com.kite9.pipeline.command.xml.replace

import com.kite9.pipeline.adl.holder.pipeline.ADLDom
import com.kite9.pipeline.command.CommandContext
import org.w3c.dom.Element

/**
 * Allows the to field to be loaded from a url.
 *
 * @author robmoffat
 */
class ReplaceTagUrl : ReplaceTag() {

    override fun getToContent(adl: ADLDom, ctx: CommandContext): Element {
        return getForeignElementCopy(adl.document, adl.uri!!, to!!, false, adl, ctx)
    }
}